visual_behavior/ophys/response_analysis/cell_metrics.py

```diff
@@ -0,0 +1,557 @@
+    if condition == 'changes':
+        metrics_table = metrics_table.merge(hit_miss_modulation_index, on='cell_specimen_id')
+    metrics_table['ophys_experiment_id'] = ophys_experiment_id
+    metrics_table['ophys_session_id'] = ophys_session_id
+
+    metrics_table = metrics_table.reset_index()
+    # metrics_table = metrics_table.melt(id_vars=['cell_specimen_id', 'ophys_experiment_id', 'ophys_session_id'])
+    metrics_table['condition'] = condition
+    metrics_table['session_subset'] = session_subset
+    metrics_table['stimuli'] = stimuli
+    metrics_table['events'] = use_events
+
+    metrics_table = metrics_table.rename(columns={'variable': 'metric'})
+
+    return metrics_table
+
+
+def get_metrics_df_filename(ophys_experiment_id, condition, stimuli, session_subset, use_events):
+    if use_events:
+        trace_type = 'events'
+    else:
+        trace_type = 'dFF'
+    filename = 'experiment_id_' + str(ophys_experiment_id) + '_' + condition + '_' + stimuli + '_' + session_subset + '_' + trace_type
+    return filename
+
+
+def get_metrics_df_filepath(ophys_experiment_id, condition, stimuli, session_subset, use_events):
+    import platform
+    if platform.system() == 'Linux':
+        save_dir = r'/allen/programs/braintv/workgroups/nc-ophys/visual_behavior/single_cell_metrics'
+    else:
+        save_dir = r'\\allen\programs\braintv\workgroups\nc-ophys\visual_behavior\single_cell_metrics'
+    filename = get_metrics_df_filename(ophys_experiment_id, condition, stimuli, session_subset, use_events)
+    filepath = os.path.join(save_dir, 'cell_metrics', filename + '.h5')
+    return filepath
+
+
+def get_cell_metrics_table_for_experiment(ophys_experiment_id, condition, stimulus, session_subset, use_events):
+    filepath = get_metrics_df_filepath(ophys_experiment_id, condition, stimulus,
+                                       session_subset, use_events)
+    metrics_table = pd.read_hdf(filepath, key='df')
+    return metrics_table
+
+
+def load_cell_metrics_table_for_experiments(ophys_experiment_ids, condition, stimulus, session_subset, use_events):
+    import visual_behavior.data_access.loading as loading
+    ophys_experiment_table = loading.get_filtered_ophys_experiment_table(release_data_only=True)
+    ophys_experiment_table = loading.add_superficial_deep_to_experiments_table(ophys_experiment_table)
+
+    problem_expts = []
+    metrics_table = pd.DataFrame()
+    if ophys_experiment_ids == 'all_experiments':
+        try:
+            metrics_table = get_cell_metrics_table_for_experiment(ophys_experiment_ids, condition, stimulus, session_subset,
+                                                                  use_events)
+        except BaseException:
+            print('problem for all experiments metrics table generation')
+    else:
+        for ophys_experiment_id in ophys_experiment_ids:
+            print(np.where(ophys_experiment_ids == ophys_experiment_id)[0][0], 'out of', len(ophys_experiment_ids))
```

Comment:
These print statements fill the screen in an annoying way. Two possible suggestions:
1) use the `end = '\r'`argument in the print statement so that each print overwrites the last
2) use the tqdm module to make a progress bar